libs/langchain/langchain/memory/chat_memory.py

```diff
@@ -37,7 +37,7 @@ def save_context(self, inputs: Dict[str, Any], outputs: Dict[str, str]) -> None:
         """Save context from this conversation to buffer."""
         input_str, output_str = self._get_input_output(inputs, outputs)
         self.chat_memory.add_messages(
-            [HumanMessage(content=input_str), AIMessage(content=output_str)]
+            [HumanMessage(content=input_str), output_str]
```

Comment:
@ww-jermaine I have been able to replicate your issue. Your PR does not make any sense, since you have just removed the AIMessage from being saved. The error is more oriented towards the parsing and how the AIMessage is stored in the output dict being used in `saved_context`